challenge-18/challenge-18.js

```diff
@@ -1,89 +1,123 @@
+  O resultado deve ser:
+  ["<li></li>", "<li></li>", "<span></span>"]
+  */
+  console.log( '\nMatch com tags HTML vazias (abertura e fechamento da tag):' );
+  htmlRegex = /(<\w+><\/\w+>)/g;
+  var tags = "<div><ul><li></li><li></li><li><span></span></li></ul></div>";
+  var emptyHtml = tags.match(htmlRegex);
+  console.log(emptyHtml);
 
-/*
-Crie uma expressão regular que faça o match com a abertura de uma tag
-HTML qualquer.
-Ex.: "<div>", "<section>", "<blockquote>".
-Use o método match e faça o teste com a marcação abaixo:
-"<div><section><blockquote>Texto <img /></blockquote></section></div>"
-O resultado deve ser:
-["<div>", "<section>", "<blockquote>"]
-*/
-console.log( '\nMatch com a abertura de uma tag HTML:' );
-// ?
+  /*
+  Vamos complicar um pouco agora :D
 
-/*
-Crie uma expressão regular que faça o match com uma tag HTML vazia, casando
-com a abertura e fechamento da tag.
-Ex.: "<div></div>", "<section></section>", "<blockquote></blockquote>".
-Use o método match e faça o teste com a marcação abaixo:
-"<div><ul><li></li><li></li><li><span></span></li></ul></div>"
-O resultado deve ser:
-["<li></li>", "<li></li>", "<span></span>"]
-*/
-console.log( '\nMatch com tags HTML vazias (abertura e fechamento da tag):' );
-// ?
+  Crie uma expressão regular que faça o match com um texto existente dentro de
+  uma tag HTML. O texto deve ser capturado e substituído por:
+  'O texto dentro da tag "[NOME DA TAG]" é "[TEXTO]"'
 
-/*
-Vamos complicar um pouco agora :D
+  Use a marcação abaixo para fazer o replace:
+  "<h1>Título da página</h1><p>Este é um parágrafo</p><footer>Rodapé</footer>"
 
-Crie uma expressão regular que faça o match com um texto existente dentro de
-uma tag HTML. O texto deve ser capturado e substituído por:
-'O texto dentro da tag "[NOME DA TAG]" é "[TEXTO]"'
+  A marcação deve permanecer como está, somente o texto deve ser substituído.
+  No replace, utilize quebras de linha para deixar uma tag por linha.
 
-Use a marcação abaixo para fazer o replace:
-"<h1>Título da página</h1><p>Este é um parágrafo</p><footer>Rodapé</footer>"
+  O resultado deve ser esse:
+  <h1>O texto dentro da tag "h1" é "Título da página"</h1>
+  <p>O texto dentro da tag "p" é "Este é um parágrafo"</p>
+  <footer>O texto dentro da tag "footer" é "Rodapé"</footer>
 
-A marcação deve permanecer como está, somente o texto deve ser substituído.
-No replace, utilize quebras de linha para deixar uma tag por linha.
+  Uma dica: faça o match aos poucos. Para facilitar o teste, use o site
+  https://regex101.com/#javascript e verifique se as capturas estão
+  corretas, para depois aplicar no código ;)
+  */
+  console.log( '\nFazer replace dos textos das tags:' );
+  htmlRegex = /<(\w+)>(.+?)</g;
+  var lastHtml = "<h1>Título da página</h1><p>Este é um parágrafo</p><footer>Rodapé</footer>";
+  var result = lastHtml.replace(htmlRegex, function (regex, m1, m2, pos) {
+    m2 = '\nO texto dentro da tag "'  + m1 + '" é "'  + m2 + '"<';
```

Comment:
@nono21, acho que só precisa entender melhor o que vai ser substuído pelo replace:

O replace substitui o que estiver no primeiro parâmetro, e ele encontrar na string, pelo segundo parâmetro, certo?

Quando usamos regex, tudo o que casar com a regex será substituído. No caso, você pega a abertura da tag inteira, o texto dentro dela e o início do fechamento da tag. É isso que será substituído.

Se você fizer a regex para pegar a tag inteira (abertura, texto dentro e fechamento), vai ficar mais fácil fazer o replace.

E os parâmetros que estão na função, você pode usar mais de uma vez para montar o que será substituído, se for o caso :)